teps/0056-pipelines-in-pipelines.md

```diff
@@ -2,283 +2,389 @@
+execution would also improve visualization of `Pipelines`.
+
+As a `Pipeline` author, I need to define a set of `Tasks` as a complete unit of execution within a `Pipeline`. For
+example, I have 5 `Tasks` that apply templates including a `DeploymentConfig`. After the 5 `Tasks` are completed,
+I have 3 other `Tasks` that add `ConfigMaps` and `Secrets` to the `DeploymentConfig`. I need to specify that the second
+set of `Tasks` all need to wait for the first set of `Tasks` to complete execution. Today, I'd have to add each of the
+5 `Task` names to the `runAfter` section of each of the 3 `Task` names - adding up to 15 items in `runAfter` that I
+have to maintain.
 
 ```
-<<[UNRESOLVED optional short context or usernames ]>>
-Stuff that is being argued.
-<<[/UNRESOLVED]>>
+                                   git-clone
+                                       |
+                                       v
+                                     build
+                                       |
+                                       v
+           ----------------------------------------------------------
+           |              |            |             |              |
+           v              v            v             v              v          
+    apply-configmap   apply-pvc   apply-route   apply-service     apply-dc
+    ----------------------------------------------------------------------
+               |                   |                       |
+               v                   v                       v   
+          add-configmap    add-columna-service       add-kafka-config      
+          ------------------------------------------------------------
+                                   |                           
+                                   v                           
+                                deploy
 ```
 
-When editing TEPS, aim for tightly-scoped, single-topic PRs to keep discussions
-focused.  If you disagree with what is already in a document, open a new PR
-with suggested changes.
+Instead, I want to define and distribute `apply-config` set of 5 `Tasks` and `add-config` set of 3 `Tasks` so that I
+can specify that the latter waits for the former to complete execution.
 
-If there are new details that belong in the TEP, edit the TEP.  Once a
-feature has become "implemented", major changes should get new TEPs.
+```                           
+       git-clone
+           |                           
+           v                           
+         build 
+           |
+           v 
+      apply-config
+           |                           
+           v                           
+       add-config 
+           |
+           v 
+        deploy
+```
 
-The canonical place for the latest set of instructions (and the likely source
-of this file) is [here](/teps/NNNN-TEP-template/README.md).
+For further information, read [issue][issue-2134] and [issue][issue-4067].
```

Comment:
NIT: this translates into:

```
For further information, read issue and issue.
```

A minor formating trouble here.

Also, the biggest pain point in https://github.com/tektoncd/pipeline/issues/4067 is the resources but I agree that the `pipelines in pipelines` indeed simplifies the YAML definition.

Reply:
fixed the formatting, and agreed that it's a resource issue but pipelines in pipelines can help with the composition